### PR TITLE
calibre: 4.23.0 -> 5.9.0

### LINKED ISSUE #106256: calibre is out-of-date (4.23 vs 5.6)
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: calibre

current version: 4.23.0
desired version: 5.6

###### Notify maintainers

maintainers: @domenkozar @pSub @AndersonTorres

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by hugolgst**
###### Motivation for this change
Updates calibre version to 5.9.0 by pinning the Python version to 3.
Calibre upgraded to Python 3 since the 5.0 release.

Fixes #106256

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -20753,11 +20753,7 @@ in
 
   calculix = callPackage ../applications/science/math/calculix {};
 
-  calibre-py2 = libsForQt5.callPackage ../applications/misc/calibre { pythonPackages = python2Packages; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:20756]: Do we want to add an alias for this to notify people that it got removed?

Comment (hugolgst) [reply pkgs/top-level/all-packages.nix:20756]: Didn't thought of that sorry, could be great yes!

Comment (hugolgst) [reply pkgs/top-level/all-packages.nix:20756]: Is that what you're thinking about? 
```nix
  calibre = libsForQt5.callPackage ../applications/misc/calibre {};

  calibre-py2 = calibre;

  calibre-py3 = calibre;
```

File: pkgs/applications/misc/calibre/default.nix [hunk 2]
```diff
@@ -57,7 +57,7 @@ mkDerivation rec {
 
   nativeBuildInputs = [ pkgconfig qmake removeReferencesTo ];
 
-  CALIBRE_PY3_PORT = builtins.toString pythonPackages.isPy3k;
+  CALIBRE_PY3_PORT = builtins.toString python3Packages.isPy3k;
```
Comment (AndersonTorres) [inline pkgs/applications/misc/calibre/default.nix:60]: It looks a bit redundant...

Comment (hugolgst) [reply pkgs/applications/misc/calibre/default.nix]: Good catch, thanks a lot

**Review by thiagokokada — APPROVED** [review]

**Comment by eduardosm** [comment]
This is failing to build because it needs sip 5. There is a [draft PR](https://github.com/NixOS/nixpkgs/pull/106550) that works, but it is currently blocked on [another PR](https://github.com/NixOS/nixpkgs/pull/108041).

**Comment by hugolgst** [comment]
I'm sorry for that mistake, is it failing to build?

**Comment by eduardosm** [comment]
It is failing: https://hydra.nixos.org/build/135047343

```
/nix/store/yi9a1fafijn0nf5m8qkmnl955vi2mncm-python3-3.8.6/bin/python3.8 -c import os; os.chdir('/build/calibre-5.9.0/build/pyqt/pictureflow'); from sipbuild.tools.build import main; main(); --verbose --no-make --qmake /nix/store/cz83r6va5cmpmncdn1gmzrx52mz7ndqn-qtbase-5.15.2-dev/bin/qmake
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'sipbuild'
```

`sipbuild` is provided by sip 5, which is not currently available in nixpkgs

**Comment by hugolgst** [comment]
Indeed, very sorry again.
Do you want me to create a revert for that?

**Comment by eduardosm** [comment]
Not necessary if https://github.com/NixOS/nixpkgs/pull/108041 gets merged soon. When that happens, I will undraft https://github.com/NixOS/nixpkgs/pull/106550 to fix the build.

**Comment by anna328p** [comment]
This is blocking rebuilds for me.

**Comment by hugolgst** [comment]
Let's create a revert PR

**Comment by cole-h** [comment]
Reverted in https://github.com/NixOS/nixpkgs/pull/109322.

**Pull request merged**